apps/meteor/client/components/message/toolbox/Toolbox.tsx

```diff
@@ -14,21 +15,25 @@ import { useRoom } from '../../../views/room/contexts/RoomContext';
 import { useToolboxContext } from '../../../views/room/contexts/ToolboxContext';
 import MessageActionMenu from './MessageActionMenu';
 
-const getMessageContext = (message: IMessage, room: IRoom): MessageActionContext => {
+const isThreadFirstMessage = (message: IThreadMessage | IThreadMainMessage): boolean => {
+	return message.renderedOnThread;
+};
+
+const getMessageContext = (message: IThreadMessage | IThreadMainMessage, room: IRoom): MessageActionContext => {
 	if (message.t === 'videoconf') {
 		return 'videoconf';
 	}
 	if (isRoomFederated(room)) {
 		return 'federated';
 	}
-	if (isThreadMessage(message) || message.renderedOnThread) {
+	if (isThreadMessage(message) || isThreadFirstMessage(message)) {
 		return 'threads';
 	}
 	return 'message';
 };
 
 type ToolboxProps = {
-	message: IMessage;
+	message: IThreadMessage | IThreadMainMessage;
```

Comment:
To avoid these kind of uncertain typings, I'd recommend to also keep the `renderedOnThread` prop on the `IMessage` interface, since both of these types extend from it.

Reply:
No, ToolBox has to accept the IMessage type, all other types (IThreadMessage | IThreadMainMessage) extend the original IMessage interface, if you keep like that anyone that is reading this component will think that the ToolBox component is used just for the Thread Messages. We always need to be clear in our implementation.